main_test.go

```diff
@@ -24,20 +24,24 @@ func TestMain(m *testing.M) {
 	flag.Parse() // NOTE: this is needed to parse args from go test command
 
 	port := func() string { // Get a free port to run the server
-		listener, err := net.Listen("tcp", ":0")
+		listener, err := net.Listen("tcp", "localhost:0")
 		if err != nil {
 			log.Fatalf("failed to listen: %v", err)
 		}
 		defer listener.Close()
-		addr := listener.Addr().(*net.TCPAddr)
+		listenAddr := listener.Addr()
+
+		addr, ok := listenAddr.(*net.TCPAddr)
+		if !ok {
+			log.Fatalf("unexpected address type: %T %+v", listenAddr, listenAddr)
+		}
```

Comment:
I don't think this kind of error handling is needed. 
As far as i know, Addr returns udp or tcp. Since we starts server with tcp, we can safely assumes it is tcp addr or failed earlier in net.Listen line